projectq/cengines/_replacer/_replacer.py

```diff
@@ -153,9 +153,25 @@ def _process_command(self, cmd):
                 if d.check(cmd):
                     decomp_list.append(d)
 
+            # If nothing found so far, check recursively parent classes:
             if len(decomp_list) == 0:
-                raise NoGateDecompositionError("\nNo replacement found for "
-                                               + str(cmd) + "!")
+                for parent_class in type(cmd.gate).mro()[1:-1]:
```

Comment:
I think it would be nice to combine the loop with the code above (i.e. loop over all classes returned by mro() and also check for inverse rules).

Reply:
It now also checks all the parent classes of the inverse gate which might only be DaggeredGate and BasicGate so the loop is a bit more complicated than initially thought.